### ISSUE TITLE
Investigate problems with using the system on the Mac and iDevices

### ISSUE THREAD

**Issue by mohammad-suliman** [hasadna/sensory-interface#58]
We got reports from users having a problems with using the system on the Mac:
* The user keeps hearing the instruction about turning off the speech when navigating the braille view area of the graph with arrow keys.
* Braille disappears for a while also when navigating the braille view area.
* The system was also initially tested on the iPhone with a Bluetooth braille display and was found to have issues.
* The Mac issues are not verified yet.
* The problems on the iPhone needs more investigation.

**Comment by mohammad-suliman** [comment]
Updates:
* The braille was found to not function properly on iOS 12. however, it functions as expected on iOS 13, where Apple introduced significant enhancements for braille.
* We need to pay attention that the cursor behaves differently on iOS, iPadOS and the Mac. This is important for correctly reporting the value under the cursor in speech and tones in the braille controller.

**Comment by S0AndS0** [comment]
A web-search about this issue resulted in [Cross browser Javascript code to get/set caret position in textarea/input](https://www.vishalon.net/blog/javascript-getting-and-setting-caret-position-in-textarea) blog post by Vishal Monpara


## TLDR code snip...


```JavaScript
function getCaretPosition(ctrl) {
    // IE < 9 Support 
    if (document.selection) {
        ctrl.focus();
        var range = document.selection.createRange();
        var rangelen = range.text.length;
        range.moveStart('character', -ctrl.value.length);
        var start = range.text.length - rangelen;
        return {
            'start': start,
            'end': start + rangelen
        };
    } // IE >=9 and other browsers
    else if (ctrl.selectionStart || ctrl.selectionStart == '0') {
        return {
            'start': ctrl.selectionStart,
            'end': ctrl.selectionEnd
        };
    } else {
        return {
            'start': 0,
            'end': 0
        };
    }
}
 
function setCaretPosition(ctrl, start, end) {
    // IE >= 9 and other browsers
    if (ctrl.setSelectionRange) {
        ctrl.focus();
        ctrl.setSelectionRange(start, end);
    }
    // IE < 9 
    else if (ctrl.createTextRange) {
        var range = ctrl.createTextRange();
        range.collapse(true);
        range.moveEnd('character', end);
        range.moveStart('character', start);
        range.select();
    }
}
```


... perhaps with a bit of TypeScript translation some of the above code could be Incorporated.

**Comment by oferb** [comment]
@S0AndS0!! We missed you!

**Comment by S0AndS0** [comment]
@oferb, the feeling is certainly mutual ;-) I'll attempt to swing through more once caught-up on recent code changes.

**Comment by oferb** [comment]
Decreasing priority as our focus now is on Windows users first.
